packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -1178,8 +1178,11 @@ class _DropdownButtonState<T> extends State<DropdownButton<T>> with WidgetsBindi
   TextStyle get _textStyle => widget.style ?? Theme.of(context).textTheme.subtitle1;
 
   void _handleTap() {
+    final RenderBox overlayBox = Overlay.of(context).context.findRenderObject() as RenderBox;
+    final Offset overlayOffset = overlayBox.localToGlobal(Offset.zero);
```

Comment:
Instead of converting both to the global coordinate system, wouldn't it be enough to convert the itemBox to the coordinate system of the overlay by providing an ancestor argument to localToGlobal?

Reply:
What about this?